### DISCUSSION TITLE
volume server create too many files

### DISCUSSION THREAD

**Discussion by banane77** [seaweedfs/seaweedfs#3877] [Q&A]
My seaweedfs volume server are managed by systemd. When the volume server process's open files reach the limit, the volume server print out lots of logs like this:
![image](https://user-images.githubusercontent.com/19773796/196376988-0d97cd53-ce7c-49bb-b21d-f53c152be139.png)
but the volume server DO NOT stop to assign new volume. so a lots of .ldb and .vif files created in the data directory, more than 1200000. And the volumeId reached 999999.
![image](https://user-images.githubusercontent.com/19773796/196378729-f3df1b36-4507-4b7b-a8bf-8ad8e54f82af.png)

Most of .ldb files are 18bytes and most of .vif files are 57 bytes. 
When I try to add the following config to systemd service file and try to reboot the volume server ,it start failed.
LimitCORE=infinity
LimitNOFILE=2024000
LimitNPROC=2024000
The volume server try to load all the volumes so the kernel killed it because of OOM. Can I delete the raw volume files to decrease the volume numbers ?
How should I do to deal with this situation ??  Is any good idea to clean the empty volume? Please Help! 
Thanks.

**Comment by chrislusf** [comment]
You can directly "rm" these files.

**Comment by chrislusf** [comment]
Need some ideas on how to improve here. The number of open files seems not able to detect via code.

**Comment by banane77** [comment]
Is this picture helpful? This error code is printed on line 48 of volume_grpc_admin.go , detail message contains "too many open files”.

I am not familiar with go language, Is this exception able to be caught when deals with system IO ?

Zhao Stephen
***@***.***
